src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -47,6 +47,8 @@ internal sealed class PrivateImplementationDetails : DefaultTypeDef, Cci.INamesp
             new ConcurrentDictionary<ImmutableArray<byte>, MappedField>(ByteSequenceComparer.Instance);
 
         private ModuleVersionIdField _mvidField;
+        // Dictionary that maps from analysis kind to instrumentation payload field.
+        private readonly ConcurrentDictionary<int, InstrumentationPayloadField> _instrumentationPayloadFields = new ConcurrentDictionary<int, InstrumentationPayloadField>();
```

Comment:
It feels like this one will be used only in limited scenarios, it might worth optimizing for this.